crates/accelerate/src/convert_2q_block_matrix.rs

```diff
@@ -48,6 +50,12 @@ pub fn blocks_to_matrix(
         [] => Mat::<c64>::identity(4, 4),
         _ => unreachable!(),
     };
+
+    let mut aux = Mat::<c64>::with_capacity(4, 4);
+    // SAFETY: `aux` is a 4x4 matrix whose values are uninitialized and it's used only to store the
+    // result of the `matmul` call inside the for loop
+    unsafe { aux.set_dims(4, 4) };
```

Comment:
This matrix is used to avoid having to clone the result of the matrix multiplication in each iteration of the loop. The result will be stored initially on `aux`, and using a `swap` operation (line 81), the result will be moved to `matrix: Mat<c64>` to be ready for the next iteration or to finish the loop and continue with the code.